pkg/products/rhssocommon/reconciler.go

```diff
@@ -280,9 +269,56 @@ func (r *Reconciler) SetupOpenshiftIDP(ctx context.Context, serverClient k8sclie
 			},
 		})
 	}
+
+	return nil
+}
+
+// Why do we need to do this?
+func (r *Reconciler) SyncOpenshiftIDPClientSecret(ctx context.Context, serverClient k8sclient.Client, authenticated keycloakCommon.KeycloakInterface, sso config.RHSSOInterface, keycloakRealmName string) error {
+
+	clientSecret, err := r.getClientSecret(ctx, serverClient, sso)
+	if err != nil {
+		return err
+	}
+
+	idp, err := authenticated.GetIdentityProvider(idpAlias, keycloakRealmName)
+	if err != nil {
+		return fmt.Errorf("failed to get identity provider via keycloak api %w", err)
+	}
+
+	idp.Config["clientSecret"] = clientSecret
```

Comment:
Are we sure this point, the map is created? This operation would fail in an empty map.

Reply:
Fixed